pkg/controller/deployment/deployment_helper.go

```diff
@@ -76,6 +76,74 @@ func parseArgMap(argMap map[string]string, args []string) {
 	}
 }
 
+// mergeContainerResources merges source container resources with that
+// provided as override resources.
+func mergeContainerResources(sourceResources corev1.ResourceRequirements, overrideResources corev1.ResourceRequirements) corev1.ResourceRequirements {
+	mergedResourceLimits := mergeContainerResourceList(sourceResources.Limits, overrideResources.Limits)
+	mergedResourceRequests := mergeContainerResourceList(sourceResources.Requests, overrideResources.Requests)
+	mergedResourceClaims := mergeResourceClaims(sourceResources.Claims, overrideResources.Claims)
+
+	return corev1.ResourceRequirements{
+		Limits:   mergedResourceLimits,
+		Requests: mergedResourceRequests,
+		Claims:   mergedResourceClaims,
+	}
+}
+
+// mergeContainerResourceList merges source resource list with that
+// provided as override resource list.
+func mergeContainerResourceList(sourceResourceList corev1.ResourceList, overrideResourceList corev1.ResourceList) corev1.ResourceList {
+	if sourceResourceList == nil && overrideResourceList == nil {
+		return nil
+	}
+
+	destResourceList := corev1.ResourceList{}
+	parseResourceList(destResourceList, sourceResourceList)
+	parseResourceList(destResourceList, overrideResourceList)
+
+	return destResourceList
```

Comment:
This feels redundant as `ResourceList` is already a mad[resourceName]Quantity. Why can't we just explicitly set values for the `Memory` and `CPU`?

Reply:
```suggestion
	if sourceResourceList == nil && overrideResourceList == nil {
		return nil
	}

	destResourceList := corev1.ResourceList{}
	if !sourceResourceList[corev1.ResourceCPU].IsZero() {
	        destResourceList[corev1.ResourceCPU] = sourceResourceList[corev1.ResourceCPU].DeepCopy()
	}
	
	if !sourceResourceList[corev1.ResourceMemory].IsZero() {
	        destResourceList[corev1.ResourceMemory] = sourceResourceList[corev1.ResourceMemory].DeepCopy()
	}

	return destResourceList
```